tests/unit/test_client.py

```diff
@@ -250,18 +254,44 @@ def test_instance_admin_api(self, mock_em):
         credentials.with_scopes.assert_called_once_with(expected_scopes)
 
     @mock.patch("google.cloud.spanner_v1.client._get_spanner_emulator_host")
-    def test_instance_admin_api_emulator(self, mock_em):
+    def test_instance_admin_api_emulator_env(self, mock_em):
```

Comment:
I can see there are tests for instance and admin clients. But how about the spanner data clients?

Reply:
The emulator host is passed as an argument to the instance. This means for spanner there is no difference between using the flag or using AnonymousCredentials. There is an existing test for setting emulator_host: `test_spanner_api_w_emulator_host`